src/utilities/viewer.cpp

```diff
@@ -217,15 +215,29 @@ class TestViewerWindow final : private D3D11Resources {
       .lpszClassName = CLASS_NAME,
     };
     RegisterClass(&wc);
+
+    POINT TOP_LEFT = POINT(mSettings.mWindowX, mSettings.mWindowY);
+    POINT BOTTOM_RIGHT = POINT(
+      TOP_LEFT.x + mSettings.mWindowWidth,
+      TOP_LEFT.y + mSettings.mWindowHeight);
+
+    HMONITOR MONA = MonitorFromPoint(TOP_LEFT, MONITOR_DEFAULTTONULL);
+    HMONITOR MONB = MonitorFromPoint(BOTTOM_RIGHT, MONITOR_DEFAULTTONULL);
+
+    if (MONA == NULL || MONA != MONB) {
+      mSettings.mWindowX = CW_USEDEFAULT;
+      mSettings.mWindowY = CW_USEDEFAULT;

```

Comment:
- width and height should also be set in this case
- to avoid duplicating defaults:

```C++
const ViewerSettings defaults {};
mSettings.mWindowX = defaults.mWindowX;
// ... and ditto for the others
```

Reply:
I disagree with overriding the size of the window, that's the whole point of this change.. the size of the window is reflected in OBS, if that changes back for whatever reason, the scene gets messed up and it'd need a manual fix.

Reply:
I was wanting this so that both corners are on the same monitor; we can solve both problems (most of the time), but it's going to be a more code:

1. If saved position fits on one monitor, use that, done.
2. If not, "center" it on the primary monitor, using the "45% of space at top, 55% of space at bottom" definition of "center" [from the MS guidelines](https://learn.microsoft.com/en-us/windows/win32/uxguide/win-window-mgt). If both corners fit on the same monitor, use that, done.
3. If not, use `GetSystemMetrics()` with `SM_CXMAXIMIZED` and `SM_CYMAXIMIZED` to get the max usable size; adjust the saved size to fit within that, preserving aspect ratio (you might want to use `OpenKneeboard::PixelSize::IntegerScaledToFit(..., ShrinkOnly)`)
4. Repeat step 2, but using the size from Step 3

With this, size should be preserved as long as it fits on the primary monitor. Could also make it go to a non-primary monitor if it's bigger, but that's a fair bit more involved.